src/input.hpp

```diff
@@ -110,13 +108,16 @@ struct TouchInput final : public Input, public EventListener {
     };
 
     std::unordered_map<SDL_FingerID, tl::optional<PressedState>> m_finger_state;
+    std::vector<SDL_Event> m_event_buffer;
 
 
 public:
     explicit TouchInput(Tetrion* target_tetrion, OnEventCallback on_event_callback = OnEventCallback{})
-        : Input{ target_tetrion, std::move(on_event_callback) } { }
+        : Input{ target_tetrion, std::move(on_event_callback) },
+          m_event_buffer{ std::vector<SDL_Event>{} } { }
```

Comment:
This also shouldn't be necessary (see other comment).
```suggestion
        : Input{ target_tetrion, std::move(on_event_callback) } { }
```